arviz/plots/distcomparisonplot.py

```diff
@@ -134,6 +135,22 @@ def plot_dist_comparison(
         for data, var in zip(datasets, var_names)
     ]
 
+    total_plots = sum(
+        sum(1 for _ in xarray_var_iter(data, var_names=var, combined=True))
+        for data, var in zip(datasets, var_names)
+    )
+    maxplots = sum(len(splot) for splot in dc_plotters)
+
+    if total_plots > rcParams["plot.max_subplots"]:
+        warnings.warn(
+            "rcParams['plot.max_subplots'] ({max_plots}) is smaller than the number "
+            "of variables to plot ({len_plotters}), generating only {max_plots} "
+            "plots".format(max_plots=maxplots, len_plotters=total_plots),
+            UserWarning,
+        )
+        for i in range(0, len(dc_plotters)):
+             dc_plotters[i] = dc_plotters[i][:len_plots]
```

Comment:
This is not necessary because the inner lists are truncated at creation, this would have no effect.